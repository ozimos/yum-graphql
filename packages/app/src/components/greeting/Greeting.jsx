import React from 'react';
import PropTypes from 'prop-types';

const Greeting = props =>
  (
    <div>
      <h4 className="greeting">
        Welcome {props.isCaterer ? 'Caterer' : 'Customer'} {props.firstName}
      </h4>
    </div>
  );
Greeting.propTypes = {
  isCaterer: PropTypes.bool.isRequired,
  firstName: PropTypes.string.isRequired
};
export default Greeting;
