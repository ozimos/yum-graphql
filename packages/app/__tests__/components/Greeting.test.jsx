/* global React:false, shallow:false */

import Greeting from '../../src/components/greeting/Greeting';

const props = {
  isCaterer: true,
  firstName: 'alias'
};
const props2 = {
  isCaterer: false,
  firstName: 'alias'
};
describe('Login Component', () => {

  it('renders correctly', () => {
    let wrapper = shallow(<Greeting {...props} />);
    expect(wrapper).toMatchSnapshot();
    wrapper = shallow(<Greeting {...props2} />);
    expect(wrapper).toMatchSnapshot();
  });

});

