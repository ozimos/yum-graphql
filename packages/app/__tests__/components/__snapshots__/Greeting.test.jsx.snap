// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<div>
  <h4
    className="greeting"
  >
    Welcome 
    Caterer
     
    alias
  </h4>
</div>
`;

exports[`Login Component renders correctly 2`] = `
<div>
  <h4
    className="greeting"
  >
    Welcome 
    Customer
     
    alias
  </h4>
</div>
`;
